### PR TITLE
[Opaque pointer] Port sycl-post-link and SPIRITTAnnotations tests to opaque pointers

### PR DESCRIPTION
**Pull request by jsji**
- Port sycl-post-link to opaque pointers
- Update SPIRITTAnnotations tests to opaque pointers

### CODE REVIEW

File: llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll [hunk 1]
```diff
@@ -64,24 +64,24 @@
 ; CHECK-LABEL: define dso_local void @_Z4testv
 define dso_local void @_Z4testv() local_unnamed_addr #0 {
 entry:
-  %call.i = tail call fast half @_Z37__sycl_getScalar2020SpecConstantValueIDhET_PKcPvS3_(i8* getelementptr inbounds ([35 x i8], [35 x i8]* @__builtin_unique_stable_name._Z27get_specialization_constantIL_Z9id_halfE17specialization_idIdEdET1_v, i64 0, i64 0), i8* bitcast (%class.specialization_id* @id_half to i8*), i8* null)
-; CHECK-DEF: %[[GEP:[0-9a-z]+]] = getelementptr i8, i8* null, i32 0
-; CHECK-DEF: %[[BITCAST:[0-9a-z]+]] = bitcast i8* %[[GEP]] to half*
-; CHECK-DEF: %[[LOAD:[0-9a-z]+]] = load half, half* %[[BITCAST]], align 2
+  %call.i = tail call fast half @_Z37__sycl_getScalar2020SpecConstantValueIDhET_PKcPvS3_(ptr getelementptr inbounds ([35 x i8], ptr @__builtin_unique_stable_name._Z27get_specialization_constantIL_Z9id_halfE17specialization_idIdEdET1_v, i64 0, i64 0), ptr @id_half, ptr null)
+; CHECK-DEF: %[[GEP:[0-9a-z]+]] = getelementptr i8, ptr null, i32 0
+; CHECK-DEF: %[[BITCAST:[0-9a-z]+]] = bitcast ptr %[[GEP]] to ptr
```
Comment (jsji) [inline llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll:69]: Not sure why we still have bitcast here, can someone more familiar with sycl-post-link have a look?

**Review by MrSidims — APPROVED** [review]

File: llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll (hunk 1 above)
Comment (MrSidims) [reply llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll:69]: Probably if we have an explicit creation of bitcast it would remain

**Comment by jsji** [comment]
Thanks @MrSidims !

@intel/llvm-gatekeepers  Can we get this merged, thanks!

File: llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll (hunk 1 above)
Comment (maksimsab) [reply llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll:69]: Yes. That is correct.

**Pull request merged**